detoxify/detoxify.py

```diff
@@ -13,21 +13,25 @@
 
 
 def get_model_and_tokenizer(
-    model_type, model_name, tokenizer_name, num_classes, state_dict
+    model_type, model_name, tokenizer_name, num_classes, state_dict, pretrained_model_path=None
 ):
     model_class = getattr(transformers, model_name)
     model = model_class.from_pretrained(
-        pretrained_model_name_or_path=None,
-        config=model_type,
+        pretrained_model_name_or_path=pretrained_model_path,
         num_labels=num_classes,
         state_dict=state_dict,
+        local_files_only=pretrained_model_path is not None,
+    )
+    tokenizer = getattr(transformers, tokenizer_name).from_pretrained(
+        pretrained_model_path or model_type,
+        local_files_only=pretrained_model_path is not None,
+        model_max_length=512,
```

Comment:
@Borda don't see any reason for hardcoding 512 here since different models can have different default values?

Reply:
I do not have good explanation, but without it or using lower number like 256 it fails

Reply:
Hm it seems to work fine for me in both cases, could you share the error you get?

Reply:
I think I do not have the failer in kernel output, it was while the interactive session...
Pls, have you tried your suggestion on Kaggle with the internet off? :rabbit:

Reply:
Sounds like this is something that's specific to how you're running it on Kaggle then since it works offline outside of it, so still don't really see the reason to hardcode it here

Reply:
I do not mind dropping it but that would be fair, as you say, to mention that this implementation may not work on Kaggle :]

Reply:
```suggestion
        # TODO: may be needed to let it work with Kaggle competition
        # model_max_length=512,
```